pkg/volume/volume.go

```diff
@@ -335,33 +336,32 @@ func (params *Parameters) ToResourceGroupModify(rg *lapi.ResourceGroup) (lapi.Re
 // It will select the flag matching the top-most layer that supports disk-less resources.
 func (params *Parameters) DisklessFlag() (string, error) {
 	for _, l := range params.LayerList {
-		if l == lapi.DRBD {
+		switch l {
+		case devicelayerkind.Drbd:
 			return lc.FlagDrbdDiskless, nil
-		}
-		if l == lapi.NVME {
-			return lc.FlagNvmeInitiator, nil
-		}
-		if l == lapi.OPENFLEX {
-			// Openflex volumes are connected via NVMe
+		case devicelayerkind.Nvme, devicelayerkind.Openflex:
 			return lc.FlagNvmeInitiator, nil
+		case devicelayerkind.Exos, devicelayerkind.Writecache, devicelayerkind.Cache, devicelayerkind.Storage, devicelayerkind.Luks:
```

Comment:
maybe just a default case? we do not really get a benefit if we enumerate all of them here, or do we? like when we add a new one we don't get some warning that it is not handled or anything, right?

Reply:
Tell that to the linter ;-)

Reply:
Replaced by `default:` + setting `default-signifies-exhaustive: true` on the linter. Since we only check "new" issues, this linter would go out-of-date really quickly anyways the next time someone updates golinstor.